### PR TITLE
Ds1 refactor: test bugs + descriptive errors + SetNumberOfWall/FloorLayers

### PR DESCRIPTION
**Pull request by gucio321**
Hi there,
I've fixed some test bugs
also I've added more descriptive messages for errors returned by DS1.LoadDS1

### CODE REVIEW

**Comment by gucio321** [comment]
in my opinion no need for `SetNumberOfShadowLayers` (because always 1)

File: d2common/d2fileformats/d2ds1/ds1.go [hunk 1]
```diff
@@ -323,6 +322,38 @@ func (ds1 *DS1) NumberOfWallLayers() int {
...
+	for y := range ds1.tiles {
+		for x := range ds1.tiles[y] {
+			newWalls := make([]Wall, n)
+			for v := int32(0); v < n; v++ {
+				newWalls[v] = ds1.tiles[y][x].Walls[v]
+			}
+
+			ds1.tiles[y][x].Walls = newWalls
+		}
+	}
```
Comment (gravestench) [inline d2common/d2fileformats/d2ds1/ds1.go:352]: Wont this explode if you are setting the number of walls to a greater value than existing wall count?

Suppose there is only 1 wall layer, and you call `ds1.SetNumberOfWallLayers(3)`:
as `v` increments towards `n`, when `v == 2` is true, `ds1.tiles[y][x].Walls[v]` will throw index out of range error.

**Review by gravestench — APPROVED** [review]

**Comment by gravestench** [comment]
Adding this at the end of `TestDS1_SetNumberOfFloors` will throw an error:
```golang
	newNumber = 10

	ds1.SetNumberOfFloorLayers(newNumber)

	test = func(ds1 *DS1) {
		if len(ds1.tiles[0][0].Floors) != int(newNumber) {
			t.Fatal("unexpected floors length set")
		}

		if ds1.numberOfFloorLayers != newNumber {
			t.Fatal("unexpected floors length set")
		}
	}

	if err := testIfRestorable(ds1, test); err != nil {
		t.Errorf("unable to restore: %v", err)
	}
```

**Comment by gucio321** [comment]
we cannot add more floor layers, than d2enum.LayerStreamFloor2

**Pull request merged**